src/views/PerPrioritizationForm/index.tsx

```diff
@@ -252,6 +273,61 @@ export function Component() {
     const handleFormSubmit = createSubmitHandler(validate, setError, handleSubmit);
     const handleFormFinalSubmit = createSubmitHandler(validate, setError, handleFinalSubmit);
 
+    const readOnlyMode = statusResponse?.phase !== STEP_PRIORITIZATION;
+
+    const handleSortChange = useCallback(
+        (newSort: SortKey) => {
+            setSortBy(newSort);
+        },
+        [],
+    );
+
+    const sortedFormComponents = useMemo(
+        () => (
+            [...formComponentResponse?.results ?? []].sort(
+                (a, b) => {
+                    if (sortBy === 'rating') {
+                        const ratingA = assessmentComponentResponseMap
+                            ?.[a.id]?.rating_details?.value ?? 0;
+                        const ratingB = assessmentComponentResponseMap
+                            ?.[b.id]?.rating_details?.value ?? 0;
+                        return compareNumber(ratingA, ratingB, -1);
+                    }
+
+                    if (sortBy === 'benchmark') {
+                        const benchmarkA = assessmentComponentResponseMap
+                            ?.[a.id]?.question_responses?.length ?? 0;
+                        const benchmarkB = assessmentComponentResponseMap
+                            ?.[b.id]?.question_responses?.length ?? 0;
+
+                        return compareNumber(benchmarkA, benchmarkB, -1);
```

Comment:
Do not need to set default value as zero.
The compareNumber function can handle that.

Reply:
When undefined, it gets more priority which is not the desired behavior

Reply:
We should also discuss about the priority of `undefined` variables.